balancer/rls/internal/keys/builder.go

```diff
@@ -123,9 +122,31 @@ func (bm BuilderMap) RLSKey(md metadata.MD, path string) KeyMap {
 	return b.keys(md)
 }
 
-// BuilderMapEqual returns true if the provided BuilderMap objects are equal.
-func BuilderMapEqual(a, b BuilderMap) bool {
-	return cmp.Equal(a, b, cmp.AllowUnexported(builder{}, matcher{}))
+// Equal reports whether bm and am represent equivalent BuilderMaps.
+func (bm BuilderMap) Equal(am BuilderMap) bool {
+	if (bm == nil) != (am == nil) {
+		return false
+	}
+	if len(bm) != len(am) {
+		return false
+	}
+
+	var keys []string
+	for k := range bm {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Why is sorting necessary?

And if we don't need to sort, it can be a nested for loop.